packages/odyssey-react/src/components/Toast/index.tsx

```diff
@@ -0,0 +1,166 @@
+/*!
+ * Copyright (c) 2021-present, Okta, Inc. and/or its affiliates. All rights reserved.
+ * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+ *
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+import React, { createContext, useContext, useState } from "react";
+import type { FunctionComponent, ReactElement, AnimationEvent } from "react";
+import { useCx, oid } from '../../utils';
+import Button from '../Button';
+
+export type ToastVariants = 'info' | 'success' | 'caution' | 'danger';
+
+export type PropsToast = {
+  /**
+   * The title to be displayed on the toast. Provides quick context; one line max.
+   */
+  title: string,
+  
+  /**
+   * Supplemental information. Be concise - less than three lines of content - as your Toast will soon vanish!
+   */
+  body?: string,
+
+ /**
+   * The visual variant to be displayed to the user.
+   * @default info
+   */
+  variant?: ToastVariants,
+  
+ /**
+   * The visual variant to be displayed to the user.
+   * @default info
+   */
+  id?: string,
+  
+  onAnimationEnd?: (event: AnimationEvent) => void,
+
+  onDismiss: () => void
+}
+
+export type PropsToastProvider = { 
+  /**
+   * Child react nodes which leverage the toast context. This is typically an entire app. 
+   */
+  children?: ReactElement | ReactElement[],
+
+  /**
+   * Callback function invoked when a toast exits the toast provider.
+   */
+  onToastExit?: (id: string) => void
+}
+
+export type StaticComponents = {
+  Provider: FunctionComponent<PropsToastProvider>
+}
+
+export type ToastObject = {
+  id?: string,
+  title: string,
+  body?: string,
+  variant?: ToastVariants
+}
+
+export const ToastContext = createContext<(obj: ToastObject) => void>(() => void 0 );
+
+/**
+ * Toasts are transient, non-disruptive messages that provide at-a-glance, 
+ * asynchronous feedback or updates.
+ * 
+ * @component
+ * @example  
+ * <Toast>...</Toast>
+ */
+const Toast: FunctionComponent<PropsToast> & StaticComponents = ({ title, body, variant = 'info', id, onDismiss, ...rest }) => {
+  const componentClass = useCx(
+    "ods-toast",
+    `is-ods-toast-${variant}`
+  );
+
+  const xid = id || oid();
+
+  return (
+    <aside role="status" id={xid} className={componentClass} {...rest}>
```

Comment:
we'll need to omit these rest props prior to applying them into this component! Also we should place them first so that we don't allow `role="status"` to be overriden incorrectly

Reply:
Fixed!